### PR TITLE
Updates on SiStrip G2 and calibtree

### PR DESCRIPTION
**Pull request by mdelcourt**
Upon request of the Strip Calibration and Local Reconstruction group:
Fixed bug in SiStripGainFromCalibTree.cc preventing G2 production on CalibTrees.
Updated scripts producing G2 payloads and plot production
Updated scripts producing Calibration trees

### CODE REVIEW

**Comment by arunhep** [comment]
@mdelcourt this PR #20010 also touches one of the files which you modified as well : CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc
so please be careful about the conflicts. Once #20010 will be merged then you might need to rebase, or the other way around.

also, if you want to be used in data-taking then please prepare a backport for 92X as well.

**Comment by mmusich** [comment]
Hi @arunhep as far as I can see there is a single line changed here in common between the two PRs. Chances are git will be able to resolve the conflict ?

**Comment by davidlange6** [comment]
not always - you can just merge them into a working area to find out for sure..

**Comment by mmusich** [comment]
I am not sure if that is what happens really behind the scenes, but both:
```
cmsrel CMSSW_9_3_X_2017-08-01-2300
cd CMSSW_9_3_X_2017-08-01-2300/src/
cmsenv
git cms-merge-topic 20019
git cms-merge-topic 20010
```
and 
```
cmsrel CMSSW_9_3_X_2017-08-01-2300
cd CMSSW_9_3_X_2017-08-01-2300/src/
cmsenv
git cms-merge-topic 20010
git cms-merge-topic 20019
```
work for me without problems.
Incidentally `git cms-merge-topic 20019` checks out the ~ entire CMSSW

**Comment by arunhep** [comment]
please test

**Comment by arunhep** [comment]
+1
changes seen in HLT directory for wf1000 comes from other commits. Most likely from this one https://github.com/cms-sw/cmssw/commit/d753374cd1b8a4a499318cc92b1119aac8ba2dfe

File: CalibTracker/SiStripChannelGain/test/7TeVData/certif.json [hunk 1]
```diff
@@ -0,0 +1 @@
...pNoise_Apvcycle&quot; plot is empty, pixel: very low stats.", "qual": "GOOD"}, "283407": {"comment": "", "qual": "GOOD"}, "274146": {"comment": "All looks fine, strip: All looks fine. TID -2 efficiency value is a known issue and within expected range., pixel: All looks fine", "qual": "GOOD"}, "276456": {"comment": "all Strip partition are NOT_READY (HV_OFF), strip: not in DAQ, pixel: Pixel OFF most the time, histograms empty", "qual": "BAD "}, "295342": {"comment": "Following Whiteboard instructions, strip: SiStrip: 99.3%, pixel: Following Whiteboard instructions", "qual": "BAD,"}, "274142": {"comment": "HV and GT vs lumi is false for entirety of the run., strip: HV and GT vs lumi is false for entirety of the run., pixel: HV and GT vs lumi is false for entirety of the run.", "qual": "BAD "}, "299369": {"comment": ", strip: SiStrip:99.1%", "qual": "GOOD"}, "299368": {"comment": ", strip: SiStrip:99.2%", "qual": "GOOD"}, "299450": {"comment": ", strip: SiStrip: 99.2%;", "qual": "GOOD"}, "296089": {"comment": "Whiteboard instructions, pixel: Whiteboard instructions", "qual": "BAD,"}, "296980": {"comment": ", strip: Sistrip:99.1% TID/2-94.96%\tTID/2+:93.23%", "qual": "GOOD"}, "296668":
```
Comment (mdelcourt) [inline CalibTracker/SiStripChannelGain/test/7TeVData/certif.json]: Dear @davidlange6, 
You are right, this information can (and is during the production) derived from other tools and I shouldn't have added in the pull-request in the first place.
I will correct it.
Thanks for pointing this out.

**Comment by lpernie** [comment]
please test

**Comment by franzoni** [comment]
+1

**Comment by davidlange6** [comment]
hi @mdelcourt - can the other json file(s) be removed too?

**Comment by mdelcourt** [comment]
Hi @davidlange6,
Sorry about this, I forgot to remove it in the previous commit... It has now been done.

**Comment by franzoni** [comment]
please test

**Comment by arunhep** [comment]
code-checks

**Comment by arunhep** [comment]
@mdelcourt 
code checks are rejected for this PR. Can you please implement the suggestions?
The code check logs are given above and here is the link : https://cmssdt.cern.ch/SDT/code-checks/PR-20019/515/code-checks.log

**Comment by mdelcourt** [comment]
@arunhep I applied the suggested patch.

**Comment by lpernie** [comment]
code-checks

**Comment by davidlange6** [comment]
please test

**Comment by arunhep** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**